test/e2e/test_storage.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"cloud.google.com/go/storage"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/google/uuid"
+	"google.golang.org/api/iterator"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/test/helpers"
+	"os"
+	"strings"
+	"time"
+
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	"testing"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+// makeBucket will get an existed bucket or create a new one if it is not existed
+func makeBucket(t *testing.T) string {
+	ctx := context.Background()
+	project := os.Getenv(ProwProjectKey)
+	bucketPrefix := "storage-e2e-test"
+	if project == "" {
+		t.Fatalf("failed to find %q in envvars", ProwProjectKey)
+	}
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("failed to create storage client, %s", err.Error())
+	}
+	it := client.Buckets(ctx, project)
+	bucketName := helpers.AppendRandomString(bucketPrefix)
+	for {
+		bucketAttrs, err := it.Next()
+		if err == iterator.Done {
+			// Create a new bucket if there is no qualified existing bucket
+			bucket := client.Bucket(bucketName)
+			if e := bucket.Create(ctx, project, &storage.BucketAttrs{}); e != nil {
+				t.Fatalf("failed to create bucket, %s", e.Error())
+			}
+			break
+		}
+		if err != nil {
+			t.Fatalf("failed to list buckets, %s", err.Error())
+		}
+		if strings.HasPrefix(bucketAttrs.Name, bucketPrefix) {
+			bucketName = bucketAttrs.Name
+			break
+		}
+	}
+	return bucketName
+}
+
+func getBucketHandler(t *testing.T, bucketName string) *storage.BucketHandle {
```

Comment:
it seems it is not a handler, it is a handle. `getBucketHandler` -> `getBucketHandle`. Which is also pretty funny api name. 😆